### PR TITLE
fix: add `PreviousFields` to `DeletedNode` metadata type

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

Title says it all.

### Context of Change

Example transaction where a `DeletedNode` has both `PreviousFields` and `FinalFields`: https://livenet.xrpl.org/transactions/424661CF1FD3675D11EC910CF161979553B6D135F9BD03E6F8D4611D88D27581/raw

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

### Did you update HISTORY.md?

- [x] Yes

## Test Plan

N/A

### CODE REVIEW

File: packages/xrpl/src/models/transactions/metadata.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ export interface DeletedNode {
   DeletedNode: {
     LedgerEntryType: string
     LedgerIndex: string
+    PreviousFields?: { [field: string]: unknown }
```
Comment (ckeshava) [inline packages/xrpl/src/models/transactions/metadata.ts:43]: I couldn't find the `PreviousFields` sub-section under `DeletedNode`. It only lists `LedgerWntryType`, `LedgerIndex` and `FinalFields`
Reference: https://xrpl.org/docs/references/protocol/transactions/metadata/#deletednode-fields

Only `ModifiedNode` has a sub-object titled `PreviousFields`

Comment (mvadari) [reply packages/xrpl/src/models/transactions/metadata.ts:43]: It's listed in the transaction linked above. The docs are incorrect then.

Comment (ckeshava) [reply packages/xrpl/src/models/transactions/metadata.ts:43]: Okay, I couldn't find the specific example that you are referring to -- but I agree with your premise. I verified in the code.

Comment (ckeshava) [reply packages/xrpl/src/models/transactions/metadata.ts:43]: All the `PreviousFields` appear under `ModifiedNode` in the first example of the docs page

**Review by ckeshava — APPROVED** [review]

File: packages/xrpl/src/models/transactions/metadata.ts (hunk 1 above)
Comment (mvadari) [reply packages/xrpl/src/models/transactions/metadata.ts:43]: This is the link in the PR description above: https://livenet.xrpl.org/transactions/424661CF1FD3675D11EC910CF161979553B6D135F9BD03E6F8D4611D88D27581/raw
The first deleted node has `PreviousFields` and `FinalFields`

Comment (ckeshava) [reply packages/xrpl/src/models/transactions/metadata.ts:43]: oh, I thought you were referring to the previous transaction in the docs link. Thanks, this clears it up

**Review by pdp2121 — APPROVED** [review]

**Pull request merged**